frontend/src/hooks/useGenerateReportEffect.ts

```diff
@@ -1,118 +1,324 @@
+    initReportInfosTimeout4Report(pollingIds);
+
+    const pollingQueue: Promise<IPollingRes>[] = pollingInfos.map((pollingInfo) =>
+      reportClient.polling(pollingInfo.callbackUrl),
+    );
+    const pollingResponses = await Promise.allSettled(pollingQueue);
+    const pollingResponsesWithId = assemblePollingResWithId(pollingResponses, pollingInfos);
+
+    setReportInfos((preReportInfos) => {
+      return preReportInfos.map((reportInfo) => {
+        const matchedRes = pollingResponsesWithId.find((singleRes) => singleRes.id === reportInfo.id);
+        if (!matchedRes) return reportInfo;
+
+        if (matchedRes.status === 'fulfilled') {
+          const { response } = matchedRes.value;
+          reportInfo.reportData = assembleReportData(response);
+          reportInfo.shouldShowBoardMetricsError = true;
+          reportInfo.shouldShowPipelineMetricsError = true;
+          reportInfo.shouldShowSourceControlMetricsError = true;
+        } else {
+          const errorKey = getErrorKey(matchedRes.reason, METRIC_TYPES.ALL) as keyof IReportError;
+          reportInfo[errorKey] = { message: DATA_LOADING_FAILED, shouldShow: true };
+        }
+        return reportInfo;
       });
+    });
+
+    const nextPollingInfos = getNextPollingInfos(pollingResponsesWithId, pollingInfos);
+    if (nextPollingInfos.length === 0) {
+      stopPollingReports();
+      return;
+    }
+    timerIdRef.current = window.setTimeout(() => {
+      pollingReport({ pollingInfos: nextPollingInfos, interval });
+    }, interval * 1000);
+  };
+
+  const stopPollingReports = () => {
+    window.clearTimeout(timerIdRef.current);
+    setHasPollingStarted(false);
+  };
+
+  const assembleReportData = (response: ReportResponseDTO): ReportResponseDTO => {
+    const exportValidityTime = exportValidityTimeMapper(response.exportValidityTime);
+    return { ...response, exportValidityTime: exportValidityTime };
   };
 
   const resetTimeoutMessage = (metricTypes: string[]) => {
     if (metricTypes.length === 2) {
-      setTimeout4Report(DEFAULT_MESSAGE);
+      setReportInfos((preReportInfos) => {
+        return preReportInfos.map((reportInfo) => {
+          reportInfo.timeout4Report = { message: DEFAULT_MESSAGE, shouldShow: true };
+          return reportInfo;
+        });
+      });
     } else if (metricTypes.includes(METRIC_TYPES.BOARD)) {
-      setTimeout4Board(DEFAULT_MESSAGE);
+      setReportInfos((preReportInfos) => {
+        return preReportInfos.map((reportInfo) => {
+          reportInfo.timeout4Board = { message: DEFAULT_MESSAGE, shouldShow: true };
+          return reportInfo;
+        });
+      });
     } else {
-      setTimeout4Dora(DEFAULT_MESSAGE);
+      setReportInfos((preReportInfos) => {
+        return preReportInfos.map((reportInfo) => {
+          reportInfo.timeout4Dora = { message: DEFAULT_MESSAGE, shouldShow: true };
+          return reportInfo;
+        });
+      });
```

Comment:
refactor this part, got `timeout4...` for each case, then set value for it in the `setReportInfos`

Reply:
done